aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -431,10 +431,10 @@ Unhealthy
 
 In some hosting scenarios, a pair of health checks are used that distinguish two app states:
 
-* The app is functioning but not yet ready to receive requests. This state is the app's *readiness*.
-* The app is functioning and responding to requests. This state is the app's *liveness*.
+* The app and its dependencies are functioning and able to process requests. This state is the app's *readiness*.
+* The app is functioning without checking dependencies. This state is the app's *liveness*.
```

Comment:
I don't think this is an improvement. Health checks are not one-size-fits all. It's up to the application author to understand and make decisions here - and this seems to describe guidance as being very clear cut.

I would really prefer that we define the terms clearly rather than try to turn the terms themselves into prescriptive guidance. One way to improve this might be to leave the original definitions in place and add an example.

Ex: An application needs to download a large configuration file before it is ready to process requests. We don't want the application to be rebooted if the download fails (it has retries) - so we use the liveness check to describe the liveness of the process (no additional checks). We want to prevent requests from being sent to the application before the configuration download has succeeded, so we use the readiness check to indicate "not ready" until the download succeeds.

If there's confusion around our guidance I'd suggest mining the docs in Kubernetes (where the concepts of liveness and readiness come from).  https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes

Reply:
I'll carve out some time early this week and update along those lines.

Reply:
WRT Kubernetes, that bit is in the topic and correct ...

https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-3.1#kubernetes-example

... it's just the initial definitions and purpose that are off.